go/mcap/mcap.go

```diff
@@ -214,12 +216,65 @@ type ChunkIndex struct {
 // calibration data, or other arbitrary data. Attachment records must not appear
 // within a chunk.
 type Attachment struct {
-	LogTime    uint64
-	CreateTime uint64
-	Name       string
-	MediaType  string
-	Data       []byte
-	CRC        uint32
+	LogTime       uint64
+	CreateTime    uint64
+	Name          string
+	MediaType     string
+	ContentLength uint64
+	Data          io.Reader
+}
+
+// AttachmentReader represents an attachment for handling in a streaming manner.
+type AttachmentReader struct {
+	LogTime       uint64
+	CreateTime    uint64
+	Name          string
+	MediaType     string
+	ContentLength uint64
+	Data          io.Reader
+
+	baseReader io.Reader
+	crcReader  *crcReader
+	crc        *uint32
+}
+
+// ComputedCRC discards any remaining data in the Data portion of the
+// AttachmentReader, then returns the checksum computed from the fields of the
+// attachment up to the CRC. If it is called before the data portion of the
+// reader has been fully consumed, an error will be returned.
+func (ar *AttachmentReader) ComputedCRC() (uint32, error) {
+	n, err := io.Copy(io.Discard, ar.Data)
```

Comment:
could alternatively cast this back to [LimitedReader](https://cs.opensource.google/go/go/+/go1.19.3:src/io/io.go;l=465) and check N

Reply:
yeah - I think maybe the thing to do here is convert `Data` to a private `data` typed as a LimitReader, and then expose it through a `Data() io.Reader` method. This can then avoid the cast and otherwise do as you suggest. No need to read.

Reply:
same considerations should apply to the next function as well

Reply:
added in both places